### PR TITLE
ISSUE-498 Implement FirebaseSubscription/set - update (change expiracy)

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/498

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala [hunk 1]
```diff
@@ -1520,4 +1526,289 @@ trait FirebaseSubscriptionSetMethodContract {
...
+
+    val newExpires = UTCDate(ZonedDateTime.now().plusDays(3)).asUTC.format(TIME_FORMATTER)
+
+    val updateResponse = `given`
+      .body(
+        s"""{
+           |  "using": [
+           |    "urn:ietf:params:jmap:core",
+           |    "com:linagora:params:jmap:firebase:push"],
+           |  "methodCalls": [[
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala:1543]: indent

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala [hunk 2]
```diff
@@ -1520,4 +1526,289 @@ trait FirebaseSubscriptionSetMethodContract {
...
+    assertThatJson(updateResponse).isEqualTo(
+      s"""{
+         |    "sessionState": "${SESSION_STATE.value}",
+         |    "methodResponses": [
+         |        [
+         |            "FirebaseRegistration/set",
+         |            {
+         |                "updated": {
+         |                    "${firebaseSubscription1.id.serialize}": {
+         |                      "expires": "$newExpires"
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala:1573]: As I remember, if the updated value is not diff from the input one, we will not return the updated value.

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala [hunk 3]
```diff
@@ -1361,7 +1364,9 @@ trait FirebaseSubscriptionSetMethodContract {
            |        {
            |            "update": {
            |                "${firebaseSubscription1.id.serialize}": {
-           |                  "types": ["Email"]
+           |                  "types": ["Email"],
+           |                  "expires": "$newExpires"
+           |
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala:1369]: useless line

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/FirebaseSubscription.scala [hunk 4]
```diff
@@ -184,18 +184,21 @@ case class FirebaseSubscriptionUpdateFailure(id: UnparsedFirebaseSubscriptionId,
   def asSetError: SetError = exception match {
     case e: FirebasePatchUpdateValidationException => SetError.invalidArguments(SetErrorDescription(e.error), e.asProperty)
     case e: FirebaseSubscriptionNotFoundException => SetError.notFound(SetErrorDescription(e.getMessage))
+    case e: ExpireTimeInvalidException => SetError.invalidArguments(SetErrorDescription(e.getMessage), Some(Properties.toProperties(Set("expires"))))
     case e: IllegalArgumentException => SetError.invalidArguments(SetErrorDescription(e.getMessage), None)
-    case _ => SetError.serverFail(SetErrorDescription(exception.getMessage))
-
+    case _ => {
+      // todo Logger
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/FirebaseSubscription.scala:190]: Don't forget this!

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
rebase

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**